config/defaults.go

```diff
@@ -0,0 +1,308 @@
+	MongoURIDefault          = "mongodb://localhost:27017/xapi"
+)
+
+// SQL (MySQL, PostgreSQL) config keys
+const (
+	SQLDriverDefault                    = "postgresql"
+	SQLHostDefault                      = "localhost"
+	SQLNameDefault                      = "xapi"
+	SQLPasswordDefault                  = ""
+	SQLPortDefault                      = "5432"
+	SQLReplicaDefault                   = false
+	SQLSkipInitializeWithVersionDefault = true
+	SQLTimeZoneDefault                  = "UTC"
+	SQLTxTimeoutDefault                 = "10s"
+	SQLUserDefault                      = "postgres"
+)
+
+// SQLite config keys
+const (
+	SQLiteDatabasePathDefault = "./test-json.db"
+	SQLiteSharedDefault       = true
+)
+
+// Graphql defaults
+const (
+	GraphqlEnabledDefault        = true
+	GraphqlPlaygroundPathDefault = "/graphql"
+	GraphqlServerPathDefault     = "/graphql"
+)
+
+// Monitor defaults
+const (
+	MonitorAuthTokenDefault                   = ""
+	MonitorBuxAgentURLDefault                 = "ws://localhost:8000/websocket"
+	MonitorDebugDefault                       = false
+	MonitorEnabledDefault                     = false
+	MonitorFalsePositiveRateDefault           = 0.01
+	MonitorLoadMonitoredDestinationsDefault   = false
+	MonitorMaxNumberOfDestinationsDefault     = 100000
+	MonitorMonitorDaysDefault                 = 7
+	MonitorProcessorTypeDefault               = "bloom"
+	MonitorSaveTransactionDestinationsDefault = true
+)
+
+// NewRelic defaults
+const (
+	NewRelicDomainNameDefault = "domain.com"
+	NewRelicEnabledDefault    = false
+	NewRelicLicenseKeyDefault = "BOGUS-LICENSE-KEY-1234567890987654321234"
+)
+
+// Nodes defaults
+const (
+	NodesUseMapiFeeQuotesDefault = true
+	NodesMinercraftAPIDefault    = "mAPI"
+)
+
+// NodesBroadcastClientAPIsDefault is the default template of broadcast-client api tokens
+var NodesBroadcastClientAPIsDefault = []string{"url|token"}
+
+// Notification defaults
+const (
+	NotificationsEnabledDefault         = false
+	NotificationsWebhookEndpointDefault = ""
+)
+
+// Paymail defaults
+const (
+	PaymailEnabledDefault                 = true
+	PaymailDefaultFromPaymailDefault      = "from@domain.com"
+	PaymailDefaultNoteDefault             = "bux Address Resolution"
+	PaymailDomainValidationEnabledDefault = false
+	PaymailSenderValidationEnabledDefault = true
+)
+
+// PaymailDomainsDefault by default allows for localhost domain
+var PaymailDomainsDefault = []string{"localhost"}
+
+// Redis defaults
+const (
+	RedisDependencyModeDefault        = true
+	RedisMaxActiveConnectionsDefault  = 0
+	RedisMaxConnectionLifetimeDefault = "60s"
+	RedisMaxIdleConnectionsDefault    = 10
+	RedisMaxIdleTimeoutDefault        = "10s"
+	RedisURLDefault                   = "redis://localhost:6379"
+	RedisUseTLSDefault                = false
+)
+
+// TaskManager defaults
+const (
+	TaskManagerEngineDefault    = "taskq"
+	TaskManagerFactoryDefault   = "memory"
+	TaskManagerQueueNameDefault = "development_queue"
+)
```

Comment:
Do we know, if there are other options? Maybe we could at least get rid of the Engine and Queue name choices, and just give the possibility to choose between "memory" and "redis" based task manager.

Reply:
Okay, done. I left just TaskManagerFactory option.